Sources/NIOHTTP2/HTTP2FrameParser.swift

```diff
@@ -0,0 +1,835 @@
+        // 32-bit stream identifier -- ensuring the top bit is empty
+        buf.write(integer: frame.streamID.networkStreamID!)
+        
+        func writePayloadSize(_ size: Int) {
+            var bytes: (UInt8, UInt8, UInt8)
+            bytes.0 = UInt8((size & 0xff_00_00) >> 16)
+            bytes.1 = UInt8((size & 0x00_ff_00) >>  8)
+            bytes.2 = UInt8( size & 0x00_00_ff)
+            withUnsafeBytes(of: bytes) { ptr in
+                _ = buf.set(bytes: ptr, at: start)
+            }
+        }
+        
+        let payloadStart = buf.writerIndex
+        
+        // frame payload follows, which depends on the frame type itself
+        switch frame.payload {
+        case .data(let data):
+            writePayloadSize(data.readableBytes)
+            return data
+            
+        case .headers(let headers, let priority):
+            if let priority = priority {
+                var dependencyRaw = UInt32(priority.dependency.networkStreamID ?? 0)
+                if priority.exclusive {
+                    dependencyRaw |= 0x8000_0000
+                }
+                buf.write(integer: dependencyRaw)
+                buf.write(integer: priority.weight)
+            }
+            
+            // HPACK-encoded data. Find the stream and its associated encoder/decoder
+            try self.headerEncoder.beginEncoding(allocator: self.allocator)
+            try self.headerEncoder.append(headers: headers)
+            let encoded = try self.headerEncoder.endEncoding()
+            
+            writePayloadSize(buf.writerIndex - payloadStart + encoded.readableBytes)
+            return .byteBuffer(encoded)
+            
+        case .priority(let priorityData):
+            writePayloadSize(5)
+            
+            var raw = UInt32(priorityData.dependency.networkStreamID ?? 0)
+            if priorityData.exclusive {
+                raw |= 0x8000_0000
+            }
+            buf.write(integer: raw)
+            buf.write(integer: priorityData.weight)
+            
+        case .rstStream(let errcode):
+            writePayloadSize(4)
+            buf.write(integer: UInt32(errcode.networkCode))
+            
+        case .settings(let settings):
+            writePayloadSize(settings.count * 6)
+            for setting in settings {
+                buf.write(integer: setting.parameter.networkRepresentation)
+                buf.write(integer: setting._value)
+            }
+            
+        case .pushPromise(let streamID, let headers):
+            let streamVal: UInt32 = UInt32(streamID.networkStreamID ?? 0)
+            buf.write(integer: streamVal)
+            
+            try self.headerEncoder.beginEncoding(allocator: self.allocator)
+            try self.headerEncoder.append(headers: headers)
+            let encoded = try self.headerEncoder.endEncoding()
+            
+            writePayloadSize(encoded.readableBytes + 4)
```

Comment:
Missing the keyword `self` on all of the invocations of this function. :wink:

Reply:
It's not a function on `self` 😉